packages/Laravel/src/Rector/Class_/InlineValidationRulesToArrayDefinitionRector.php

```diff
@@ -0,0 +1,184 @@
+                foreach ($newRules as &$newRule) {
+                    if (!($newRule instanceof Node\Expr\BinaryOp\Concat)) {
+                        continue;
+                    }
+
+                    try {
+                        $fullString = $this->transformConcatExpression($newRule);
+                    } catch (\LogicException $exception) {
+                        continue;
+                    }
+
+                    $matchesExist = preg_match('/^exists:(\w+),(\w+)$/', $fullString, $matches);
+                    if ($matchesExist === false || $matchesExist === 0) {
+                        continue;
+                    }
+
+                    $newRule = new Node\Expr\StaticCall(
+                        new Node\Name('\Illuminate\Validation\Rule'),
+                        'exists',
+                        [
+                            new Node\Arg(
+                                new Node\Expr\ClassConstFetch(
+                                    new Node\Name($matches[1]),
+                                    'class'
+                                )
+                            ),
+                            new Node\Arg(new Node\Scalar\String_($matches[2])),
+                        ]
+                    );
+                }
+
+                $item->value = new Node\Expr\Array_(
+                    array_map(function (Node\Expr $rule): Node\Expr\ArrayItem {
+                        return new Node\Expr\ArrayItem($rule);
+                    }, $newRules)
+                );
+            }
+        }
+
+        return $node;
+    }
+
+    /**
+     * @param Node\Expr $expr
+     * @return Node\Expr[]
+     */
+    private function transformRulesSet(Node\Expr $expr): array
+    {
+        if ($expr instanceof Node\Scalar\String_) {
+            $parts = preg_split('/\|/', $expr->value);
+            if ($parts === false) {
+                throw new \InvalidArgumentException("Failed to split string {$expr->value} with regex");
+            }
+
+            return array_map(static function (string $value): Node\Scalar\String_ {
+                return new Node\Scalar\String_($value);
+            }, $parts);
+        } elseif ($expr instanceof Node\Expr\BinaryOp\Concat) {
+            $left = $this->transformRulesSet($expr->left);
+            $expr->left = $left[count($left ) - 1];
+
+            $right = $this->transformRulesSet($expr->right);
+            $expr->right = $right [0];
+
+            return array_merge(array_slice($left, 0, -1), [$expr], array_slice($right, 1));
+        } elseif ($expr instanceof Node\Expr\ClassConstFetch
+            || $expr instanceof Node\Expr\MethodCall
+            || $expr instanceof Node\Expr\FuncCall
+        ) {
+            return [$expr];
+        }
+
+        throw new \InvalidArgumentException('Unexpected call ' . get_class($expr));
+    }
+
+    private function transformConcatExpression(Node\Expr\BinaryOp\Concat $expression): string
```

Comment:
What kind of "transformation"? More precise naming would help to understand the content